researchstack-sdk/src/main/java/org/sagebionetworks/bridge/researchstack/BridgeDataProvider.java

```diff
@@ -532,20 +541,47 @@ public void uploadTaskResult(Context context, TaskResult taskResult) {
     //       handle schedules and filters
     private SchedulesAndTasksModel translateActivities(ScheduledActivityList activityList) {
         logger.info("translateActivities()");
+
+        // first, group activities by day
+        Map<Integer, List<ScheduledActivity>> activityMap = new HashMap<>();
```

Comment:
Should this be a sorted map?  Seems like have the keys ordered by day would be beneficial.